src/CoinbaseSmartWallet.sol

```diff
@@ -176,14 +177,17 @@ contract CoinbaseSmartWallet is MultiOwnable, UUPSUpgradeable, Receiver, ERC1271
     ///      to be replayed for all accounts sharing the same address across chains. E.g. This may be
     ///      useful for syncing owner changes.
     ///
-    /// @param data The `UserOperation` raw call data of the  execute.
-    function executeWithoutChainIdValidation(bytes calldata data) public payable virtual onlyEntryPoint {
-        bytes4 selector = bytes4(data[0:4]);
-        if (!canSkipChainIdValidation(selector)) {
-            revert SelectorNotAllowed(selector);
-        }
+    /// @param calls An array of calldata to use for separate self calls.
+    function executeWithoutChainIdValidation(bytes[] calldata calls) public payable virtual onlyEntryPoint {
+        for (uint256 i; i < calls.length; i++) {
+            bytes calldata call = calls[i];
+            bytes4 selector = bytes4(call[0:4]);
+            if (!canSkipChainIdValidation(selector)) {
+                revert SelectorNotAllowed(selector);
+            }
 
-        _call(address(this), 0, data);
+            _call(address(this), 0, call);
```

Comment:
Does it make sense to validate that each call `canSkipChainValidation` before making any actual calls? It means iterating through the loop twice but would probably save more gas in revert cases.